src/Microsoft.Health.Dicom.Core/Features/Query/QueryParser.cs

```diff
@@ -180,12 +213,18 @@ private bool TryParseDicomAttributeId(string attributeId, out DicomTag dicomTag)
             return succeed;
         }
 
-        private static void ValidateIfTagSupported(DicomTag dicomTag, string attributeId, QueryResource resourceType)
+        private static void ValidateIfTagSupported(DicomTag dicomTag, string attributeId, QueryResource resourceType, out CustomTagFilterDetails customTagFilterDetails, IDictionary<QueryResource, HashSet<CustomTagFilterDetails>> customTagMapping = null)
```

Comment:
out should be the last param for readability

Reply:
The optional parameters have to be added at the end of the parameter list!